src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -849,7 +990,7 @@ public bool TryReplaceIndexes(string oldIndexName, string newIndexName, bool imm
             bool lockTaken = false;
             try
             {
-                Monitor.TryEnter(_indexAndTransformerLocker, 16, ref lockTaken);
+                Monitor.TryEnter(_locker, 16, ref lockTaken);
```

Comment:
_locker is for storage operations only, and the other one is for 'not changing simultaneously the database record in raft'